exercises/concept/gotta-snatch-em-all/.meta/Exemplar.elm

```diff
@@ -1,11 +1,61 @@
-module GottaSnatchEmAll exposing (someFunction, someOtherFunction)
+module GottaSnatchEmAll exposing (..)
 
+import Set exposing (Set)
 
-someFunction : Int -> Int
-someFunction a =
-    Debug.todo "Please implement someFunction"
 
+type alias Card =
+    String
 
-someOtherFunction : Int -> Int
-someOtherFunction b =
-    Debug.todo "Please implement someOtherFunction"
+
+newCollection : Card -> Set Card
+newCollection =
+    Set.singleton
+
+
+addCard : Card -> Set Card -> ( Bool, Set Card )
+addCard card collection =
+    ( Set.member card collection, Set.insert card collection )
+
+
+tradeCard : Card -> Card -> Set Card -> ( Bool, Set Card )
+tradeCard yourCard theirCard collection =
+    ( Set.member yourCard collection && not (Set.member theirCard collection)
+    , collection |> Set.remove yourCard |> Set.insert theirCard
+    )
+
+
+removeDuplicates : List Card -> List Card
+removeDuplicates =
+    Set.fromList >> Set.toList
+
+
+extraCards : Set Card -> Set Card -> Int
+extraCards yourCollection theirCollection =
+    Set.diff yourCollection theirCollection
+        |> Set.size
+
+
+boringCards : List (Set Card) -> List Card
+boringCards collections =
+    case collections of
```

Comment:
Does this need pattern matching? It feels like it could be done using List.foldl directly, but I might be missing something ...

Reply:
It could be done with Haskell's `foldl1`, but here I need the first element in the list to be the initial value of the accumulator, so I need to open up the list first.